extras/ping/ping.go

```diff
@@ -454,6 +472,9 @@ func newExpBackoff(baseDelay time.Duration, maxExp int64) expBackoff {
 }
 
 func (p *Pinger) recvICMP(recv chan<- *packet) error {
+	if p.conn == nil {
+		panic("not initialized")
+	}
```

Comment:
Maybe you can have your own `runtimex.Assert` function? This reduces the need to write tests for these `if`s that should never be taken unless there are issues?

Reply:
good idea, done!